### ISSUE TITLE
Automated stub uploader constantly failing because types-pika already exists on PyPI

### ISSUE THREAD

**Issue by ilevkivskyi** [python/typeshed#9246]
See for example https://github.com/typeshed-internal/stub_uploader/actions/runs/3521987995/jobs/5904439698

PyPI: https://pypi.org/project/types-pika/
Ref https://github.com/python/typeshed/pull/9200

cc @srittau @JelleZijlstra @qubidt

**Comment by srittau** [comment]
It's too bad there aren't protected namespaces on PyPI. I suggest we add a field `package` or similar to `METADATA.toml` where we can put an alternative package name if necessary.

**Comment by AlexWaygood** [comment]
Is it worth temporarily removing the stubs from typeshed for now, as a stopgap measure?

**Comment by ilevkivskyi** [comment]
Btw `mypy` (and I guess other tools too) will automatically propose to install `types-xxx` (if types for `xxx` are missing). I guess we can discuss with the owner of the original package @qubidt how to "merge efforts"?

I would propose to revert that PR for now, so that automatic uploader will work. But this is probably not urgent, I guess people can wait a day or two without stub updates (or we can use manual upload action if something is urgently needed).

(also cc @JukkaL for visibility)

**Comment by srittau** [comment]
I've created a PR for the stub_uploader supporting a "stub_distribution" field if we decide to go that way: typeshed-internal/stub_uploader#76.

**Comment by qubidt** [comment]
Happy to do whatever is needed to resolve this. Sorry for stepping on your namesake in the first place

**Comment by AlexWaygood** [comment]
> Happy to do whatever is needed to resolve this. Sorry for stepping on your namesake in the first place

I definitely don't think this is your fault, @qubidt :) I don't think there's any reason for typeshed to have any intrinsic "right" to `types-*` names on PyPI, it's just become an assumption around these parts that those names were available!

**Comment by AlexWaygood** [comment]
I do agree with @ilevkivskyi that we should work out how to collaborate, though, if that's something you'd be interested in @qubidt. It seems silly to have two different sets of stubs for `pika`.

It feels slightly unfair to ask, since you "got there first", but would you be interested in contributing your stubs to typeshed, @qubidt? It looks like you have a lot more "annotation coverage" than us at the moment over at https://github.com/qubidt/types-pika, but I think our CI checks are probably slightly more robust.

**Comment by qubidt** [comment]
yes, absolutely. although I'm not exactly how to proceed. should I try to merge the annotations from [qubidt/types-pika](https://github.com/qubidt/types-pika) into typeshed and create a PR for it?

**Comment by AlexWaygood** [comment]
> yes, absolutely. although I'm not exactly how to proceed. should I try to merge the annotations from [qubidt/types-pika](https://github.com/qubidt/types-pika) into typeshed and create a PR for it?

Yes, that would be perfect! Thank you!

**Comment by AlexWaygood** [comment]
To update on the current status now: typeshed's automated stub uploader is now green again, due to the following trio of PRs by @srittau:
- https://github.com/typeshed-internal/stub_uploader/pull/79
- #9256
- #9257

This takes the pressure off, giving us more time to work on consolidating stubs for pika without worrying about the stub uploader constantly failing :)

**Comment by AlexWaygood** [comment]
The stubs in typeshed are now uploaded to PyPI under the name `types-pika-ts`. But we should continue to work on merging annotations from types-pika into the typeshed stubs, so that end users can have the best experience possible.

**Comment by donbarbos** [comment]
With the recent changes, I think the effort to merge the type annotations is effectively complete. At this point, baodrate/types-pika no longer seems to offer any clear advantages over the typeshed stubs, especially in terms of maintenance.

It might be worth considering transferring the PyPI package name, or adding a note directing users to the typeshed stubs as the recommended option.

cc @baodrate

**Comment by LincolnPuzey** [comment]
I saw the pika project adding type hints to itself: https://github.com/pika/pika/pull/1529

Not sure what impact that has here

**Comment by donbarbos** [comment]
> I saw the pika project adding type hints to itself: [pika/pika#1529](https://github.com/pika/pika/pull/1529)

Yes, you're right. `pika` has indeed added type annotations, and more importantly, a `py.typed` file (https://github.com/pika/pika/pull/1632). However, the release containing the `py.typed` file is planned for the next minor version (`1.5.0`). The current releases still don't include `py.typed` so the package isn't considered typed yet.
